Slim/Plugin/MusicMagic/Plugin.pm

```diff
@@ -331,10 +331,16 @@ sub postinitPlugin {
 
 			# don't seed from radio stations - only do if we're playing from some track based source
 			if ($seedTracks && ref $seedTracks && scalar @$seedTracks) {
-				foreach (@$seedTracks) {
-					my ($trackObj) = Slim::Schema->find('Track', $seedTracks->[0]->{id});
+				my @seedsToUse = ();
+				foreach my $seedTrack (@$seedTracks) {
+					my ($trackObj) = Slim::Schema->find('Track', $seedTrack->{id});
+					if ($trackObj) {
+						push @seedsToUse, $trackObj->path;
+					}
+				}
 
-					my $mix = getMix($client, $trackObj->path, 'track') if $trackObj;
+				if (scalar @seedsToUse > 0) {
+					my $mix = getMix($client, 'track', @seedsToUse);
```

Comment:
I'd suggest instead of sending the list of seeds, you send a reference to the list. This would then not require you to change all calls to `getMix()`: `getMix($client, \@seedsToUse, 'track')`, but you'd only have to check that param in the function itself. See further down.

Reply:
Keep in mind that other plugins, or other code might be calling `getMix()`. It's better not to change its signature.